pandas/io/tests/test_excel.py

```diff
@@ -423,7 +425,62 @@ def test_reader_converters(self):
         for path in (xls_path, xlsx_path):
             actual = read_excel(path, 'Sheet1', converters=converters)
             tm.assert_frame_equal(actual, expected)
-
+    
+    def test_reading_all_sheets(self):
+        # Test reading all sheetnames by setting sheetname to None,
+        # Ensure a dict is returned.
+        # See PR #9450
+    
+        _skip_if_no_xlrd()
+        
+        dfs = read_excel(self.multisheet,sheetname=None)
+        expected_keys = ['Alpha','Beta','Charlie']
+        tm.assert_contains_all(expected_keys,dfs.keys())
+
+    def test_reading_multiple_specific_sheets(self):
+        # Test reading specific sheetnames by specifying a mixed list 
+        # of integers and strings, and confirm that duplicated sheet
+        # references (positions/names) are removed properly.
+        
+        # Ensure a dict is returned
+        # See PR #9450
+        _skip_if_no_xlrd()
+        
+        #Explicitly request duplicates.  Only the set should be returned.
+        expected_keys = [2,'Charlie','Charlie']
+        dfs = read_excel(self.multisheet,sheetname=expected_keys)
+        expected_keys = list(set(expected_keys))
+        tm.assert_contains_all(expected_keys,dfs.keys())
+        assert len(expected_keys) == len(dfs.keys())
+
+    def test_creating_and_reading_multiple_sheets(self):
+        # Test reading multiple sheets, from a runtime created excel file
+        # with multiple sheets.
+         # See PR #9450       
+    
+        _skip_if_no_xlrd()
+        _skip_if_no_xlwt()
+               
+        def tdf(sheetname):
+            d, i = [11,22,33], [1,2,3]
+            return DataFrame(d,i,columns=[sheetname])
+        
+        sheets = ['AAA','BBB','CCC']
+        
+        dfs = [tdf(s) for s in sheets]
+        dfs = dict(zip(sheets,dfs))
+        
+        w = pd.ExcelWriter(self.multisheet2)
```

Comment:
use a temporary filepath here, see [like this](https://github.com/pydata/pandas/blob/master/pandas/io/tests/test_excel.py#L275); you cannot assume that you can write to the test dirctory (e.g. the code); travis lets you , but on a regular install you normally cant'.